### PR TITLE
Fix: IE/Edge issues

### PR DESCRIPTION
**Pull request by vinteo**
- Encode SVG data URLs
- Only load data when mounted in ListPicker

### CODE REVIEW

File: test/components/adslotUi/TreePickerComponentTest.js [hunk 1]
```diff
@@ -23,18 +24,11 @@ describe('TreePickerComponent', () => {
...
-  const runComponentWillUnmount = ({ shallowRenderer }) =>
-    shallowRenderer._instance._instance.componentWillUnmount();
-
-  const createAndMountComponent = (component, props = {}) => {
-    const shallowRenderer = TestUtils.createRenderer();
-    shallowRenderer.render(React.createElement(component, props));
-    runComponentDidMount({ shallowRenderer });
-    return shallowRenderer.getRenderOutput();
-  };
+  const {
```
Comment (camjc) [inline test/components/adslotUi/TreePickerComponentTest.js]: may as well destructure these in the import

File: test/helpers/shallowRenderHelpers.js [hunk 2]
```diff
@@ -17,8 +17,21 @@ import TestUtils from 'react-addons-test-utils';
...
-}
+};
+
+exports.runComponentDidMount = ({ shallowRenderer }) =>
+    shallowRenderer._instance._instance.componentDidMount();
+
+exports.runComponentWillUnmount = ({ shallowRenderer }) =>
+  shallowRenderer._instance._instance.componentWillUnmount();
+
+exports.createAndMountComponent = (component, props = {}) => {
```
Comment (camjc) [inline test/helpers/shallowRenderHelpers.js]: lets make this one's api match `createComponent`, maybe make a shared function that both use

File: src/components/adslotUi/ListPickerComponent.js [hunk 3]
```diff
@@ -16,9 +16,16 @@ class ListPickerComponent extends React.Component {
...
 
+    this.state = {};
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
     this.loadData();
   }
 
+  componentWillUnmount() { this._isMounted = false; }
```
Comment (omgaz) [inline src/components/adslotUi/ListPickerComponent.js]: Coverage.

File: src/components/adslotUi/ListPickerComponent.js [hunk 4]
```diff
@@ -16,9 +16,16 @@ class ListPickerComponent extends React.Component {
       'selectItem',
     ]) {this[methodName] = this[methodName].bind(this);}
 
+    this.state = {};
```
Comment (omgaz) [inline src/components/adslotUi/ListPickerComponent.js]: getInitialState?

File: src/styles/mixins/dataUriEncoder.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    $chunk: str-replace($chunk, '"', '\'');
+    $chunk: str-replace($chunk, '<', '%3C');
+    $chunk: str-replace($chunk, '>', '%3E');
+    $chunk: str-replace($chunk, '&', '%26');
+    $chunk: str-replace($chunk, '#', '%23');
+    $encoded: #{$encoded}#{$chunk};
+    $index: $index + $slice;
+  }
+  @return url('data:image/svg+xml;charset=utf8,#{$encoded}');
+}
```
Comment (omgaz) [inline src/styles/mixins/dataUriEncoder.scss]: Can we not do this in the loader with something like https://www.npmjs.com/package/svg-url-loader?

File: src/components/adslotUi/ListPickerComponent.js (hunk 4 above)
Comment (camjc) [inline src/components/adslotUi/ListPickerComponent.js]: Thats not how ES2015 classes work, check out the docs.

File: src/components/adslotUi/ListPickerComponent.js (hunk 3 above)
Comment (camjc) [inline src/components/adslotUi/ListPickerComponent.js]: Should be able to copy what I did for coverage on the treepicker

File: test/components/adslotUi/TreePickerComponentTest.js (hunk 1 above)
Comment (omgaz) [inline test/components/adslotUi/TreePickerComponentTest.js]: +1

**Comment by camjc** [comment]
Please use the svg mixin on adslot-ui/src/styles/icon.scss too, and in alexandria too.

File: src/styles/mixins/dataUriEncoder.scss (hunk 5 above)
Comment (camjc) [inline src/styles/mixins/dataUriEncoder.scss]: I'm going to change most of if not all of the icons soon anyway, that might be a good approach so I'll check it out then. This PR is a good solution for now.

File: src/components/adslotUi/ListPickerComponent.js (hunk 3 above)
Comment (vinteo) [inline src/components/adslotUi/ListPickerComponent.js]: I just got rid of it since its not used in this component

**Comment by vinteo** [comment]
@camjc @omgaz merge?

File: test/helpers/shallowRenderHelpers.js [hunk 6]
```diff
@@ -17,8 +17,25 @@ import TestUtils from 'react-addons-test-utils';
...
  * @return {Object} Shallow rendered output
  */
-export default function createComponent(component, props = {}, ...children) {
+exports.createComponent = (component, props = {}, ...children) => {
+  const shallowRenderer = exports.createShallowRenderer(component, props, ...children);
+  return shallowRenderer.getRenderOutput();
+};
+
+exports.runComponentDidMount = ({ shallowRenderer }) =>
+    shallowRenderer._instance._instance.componentDidMount();
```
Comment (camjc) [inline test/helpers/shallowRenderHelpers.js]: bonus spaces? surprised linting didn't hassle you for this.

**Comment by camjc** [comment]
Merging!

**Pull request merged**